Sources/SWBCore/Specs/ProductTypes.swift

```diff
@@ -270,12 +270,22 @@ public class ProductTypeSpec : Spec, SpecType, @unchecked Sendable {
         if producer.isApplePlatform {
             let compatibilityVersion = scope.evaluate(BuiltinMacros.DYLIB_COMPATIBILITY_VERSION)
             if !compatibilityVersion.isEmpty {
-                args += ["-compatibility_version", compatibilityVersion]
+                switch scope.evaluate(BuiltinMacros.LINKER_DRIVER) {
+                case .clang:
+                    args += ["-compatibility_version", compatibilityVersion]
```

Comment:
Any reason to prefer the driver option at all vs just using Xlinker? Does `-compatibility_version` at the clang level actually _do_ anything?

Reply:
It doesn't have any behavior I know of but I'd prefer to use driver options whenever possible for maximum compatibility